src/compressio/type_compressor.py

```diff
@@ -1,23 +1,51 @@
+from functools import singledispatch
 from typing import Type
 
 import pandas as pd
-from visions import Complex, DateTime, Float, Integer, Object, String, VisionsBaseType
+from visions import (
+    Boolean,
+    Complex,
+    DateTime,
+    Float,
+    Integer,
+    Object,
+    String,
+    VisionsBaseType,
+)
 
 from compressio.compression_algorithms import (
     compress_complex,
     compress_datetime,
     compress_float,
     compress_integer,
     compress_object,
+    compress_sparse,
 )
+from compressio.utils import compose
+
+
+@singledispatch
+def parse_func(f):
+    raise NotImplemented
+
+
+@parse_func.register
```

Comment:
This was an issue we hit with visions but parsing the registered dtype from a type annotation wasn't added until 3.7 or 3.8. I think we have to register on the annotation, i.e. `@parse_func.register(object)`